pkg/endpoint/endpoint.go

```diff
@@ -1998,7 +1998,7 @@ func (e *Endpoint) runLabelsResolver(owner Owner, myChangeRev int, blocking bool
 
 func (e *Endpoint) identityLabelsChanged(owner Owner, myChangeRev int) error {
 	if err := e.RLockAlive(); err != nil {
-		return err
+		return nil
```

Comment:
`identityLabelsChanged` is not only called in a controller and this is also misleading as there was indeed an error the caller should be aware off.

I would suggest to create an error type called `ErrNotAlive` returned by `RLockAlive` and the caller of `identityLabelsChanged` would check if the error returned is a `ErrNotAlive` or a "normal" error.

Similar to https://github.com/cilium/cilium/blob/3d5785f37393d42725e31e891e808455ee68a632/pkg/endpointmanager/errors.go#L23

Reply:
`identityLabelsChanged` is called in `runLabelsResolver`, in two places:

https://github.com/cilium/cilium/blob/b0471e8dd05cae3d5a25263615e8401708d58135/pkg/endpoint/endpoint.go#L1965-L1997

The caller is not made aware of this error because `runLabelsResolver` doesn't return any errors. Consequently, I don't see why we need to create a new error type. If there is an error resolving the identity for an endpoint, we log it. We always start the controller in `runLabelsResolver` as well. I do not think that there is a need to say that there is an error resolving an identity for an endpoint which is being deleted, because it's going to be deleted anyway. The logic behind `RLockAlive` is to avoid doing operations on endpoints which are being deleted, so I think returning no error in this case is fine because it means that we don't have to do any more operations on the endpoint.

Reply:
In `identityLabelsChanged`, the `e.RLockAlive()` is used in more than one place, does this means you will switch all of them to `return nil` if `e.RLockAlive()` returns an error?

>no error in this case is fine because it means that we don't have to do any more operations on the endpoint.

Tomorrow, or next week, or in a month, I'll use `identityLabelsChanged` for a totally different thing, how do I know the function succeeded if in error cases we are also returning nil? It does not make any sense to me.